src/requirementslib/models/requirements.py

```diff
@@ -619,6 +619,13 @@ def locked_vcs_repo(self, src_dir=None):
             vcsrepo.checkout_ref(self.ref)
         self.ref = self.get_commit_hash()
         self.req.revision = self.ref
+
+        # Remove potential ref in the end of uri after ref is parsed
+        if "@" in self.link.show_url and "@" in self.uri:
+            uri, ref = self.uri.rsplit("@", 1)
+            if ref in self.ref:
+                self.uri = uri
```

Comment:
This makes sense to me, ditching the extra ref if it matches what we check out -- thanks for the fix!